Test/ParallelPrimitives/RadixSortKernels.h

```diff
@@ -570,7 +548,119 @@ __device__ void localSort4bitMulti( int* keys, u32* ldsKeys, const int START_BIT
 	}
 }
 
+__device__ void localSort8bitMulti_shared_bin( int* keys, volatile u32* ldsKeys, const int START_BIT )
+{
+	__shared__ unsigned table[BIN_SIZE];
+
+	for( int i = threadIdx.x; i < BIN_SIZE; i += WG_SIZE )
+	{
+		table[i] = 0U;
+	}
+
+	LDS_BARRIER;
+
+	for( int i = 0; i < SORT_N_ITEMS_PER_WI; ++i )
+	{
+		const int tableIdx = ( keys[i] >> START_BIT ) & RADIX_MASK;
+		atomicAdd( &table[tableIdx], 1 );
+	}
+
+	LDS_BARRIER;
+
+	int globalSum = 0;
+	for( int binId = 0; binId < BIN_SIZE; binId += WG_SIZE * 2 )
+	{
+		unsigned* globalOffset = &table[binId];
+		const unsigned currentGlobalSum = ldsScanExclusive( globalOffset, WG_SIZE * 2 );
+		globalOffset[threadIdx.x * 2] += globalSum;
+		globalOffset[threadIdx.x * 2 + 1] += globalSum;
+		globalSum += currentGlobalSum;
+	}
+
+	LDS_BARRIER;
+
+	for( int i = 0; i < SORT_N_ITEMS_PER_WI; i++ )
+	{
+		const int tableIdx = ( keys[i] >> START_BIT ) & RADIX_MASK;
+		const int writeIndex = atomicAdd( &table[tableIdx], 1 );
+
+		ldsKeys[writeIndex] = keys[i];
+	}
+
+	LDS_BARRIER;
+
+	for( int i = 0; i < SORT_N_ITEMS_PER_WI; ++i )
+	{
+		keys[i] = ldsKeys[threadIdx.x * SORT_N_ITEMS_PER_WI + i];
+	}
+}
+
 __device__ void localSort8bitMulti( int* keys, u32* ldsKeys, const int START_BIT )
+{
+	constexpr auto N_GROUP_SIZE{ N_BINS_8BIT / ( sizeof( u64 ) / sizeof( u16 ) ) };
```

Comment:
It is purposely named "grouped" instead of "packed" since the concept is a bit different: In the packed version we encode and pack the "bin table" in on array element of a 64-bit integer whereas in the group version the "bin table" still spreads across the array.